### PR TITLE
[BTFSBTT-173]: get StoragePriceAsk, BandwidthPriceAsk, StorageTimeMin, BandwidthLimi…

### PR DESCRIPTION
**Pull request by wxue**
…t, CollateralStake from config and send to status-server;

### CODE REVIEW

File: analytics/analytics.go [hunk 1]
```diff
@@ -107,13 +112,40 @@ func Initialize(n *core.IpfsNode, BTFSVersion string) {
...
+	go dc.collectionAgent(env)
 }
 
-func (dc *dataCollection) update() {
+func (dc *dataCollection) update(env cmds.Environment) {
 	var m runtime.MemStats
 	runtime.ReadMemStats(&m)
 
+	// get config info
+	var log = logging.Logger("cmd/btfs")
```
Comment (taiyangc) [inline analytics/analytics.go]: Make this part of package and use `logging.Logger("analytics")` (ipfs convention)

File: analytics/analytics.go [hunk 2]
```diff
@@ -107,13 +112,40 @@ func Initialize(n *core.IpfsNode, BTFSVersion string) {
...
+	node, err := cmdenv.GetNode(env)
+	if err != nil {
+		log.Warning(err.Error())
+		return
+	}
+	rds := node.Repo.Datastore()
+	selfKey := ds.NewKey(fmt.Sprintf("%s%s", "/host_storage/", node.Identity.Pretty()))
```
Comment (taiyangc) [inline analytics/analytics.go]: Make key helper a shared function, `core/commands/storage.go` also used this

File: analytics/analytics.go [hunk 3]
```diff
@@ -107,13 +112,40 @@ func Initialize(n *core.IpfsNode, BTFSVersion string) {
...
+	b, err := rds.Get(selfKey)
+	if err != nil && err != ds.ErrNotFound {
+		return
```
Comment (taiyangc) [inline analytics/analytics.go]: Should not return if host storage variables are unretrievable, probably use `dc.reportHealthAlert` here

File: analytics/analytics.go (hunk 1 above)
Comment (wxue) [reply analytics/analytics.go]: good catch, thanks

File: analytics/analytics.go (hunk 2 above)
Comment (wxue) [reply analytics/analytics.go]: will do

File: analytics/analytics.go (hunk 3 above)
Comment (wxue) [reply analytics/analytics.go]: will do

File: analytics/analytics.go (hunk 1 above)
Comment (wxue) [reply analytics/analytics.go]: good catch, thanks

File: analytics/analytics.go (hunk 2 above)
Comment (wxue) [reply analytics/analytics.go]: will do

File: analytics/analytics.go (hunk 3 above)
Comment (wxue) [reply analytics/analytics.go]: will do

File: patching.sh [hunk 4]
```diff
@@ -46,11 +34,3 @@ chmod -R 777 $patchdir
 if ! patch -R -s -f --dry-run $patchdir/record.go < ./patches/go-ipns-v.0.0.1-record.go.patch 1>/dev/null; then
     patch $patchdir/record.go < ./patches/go-ipns-v.0.0.1-record.go.patch;
 fi
-
-# patching test
-patchdir=$GOPATH/pkg/mod/github.com/ipfs/interface-go-ipfs-core@v0.1.0
-cp ./patches/test/interface-go-ipfs-core.patch $patchdir
```
Comment (taiyangc) [inline patching.sh]: remove the test patch as well

File: analytics/analytics.go [hunk 5]
```diff
@@ -118,13 +124,40 @@ func Initialize(n *core.IpfsNode, BTFSVersion, hValue string) {
...
+	go dc.collectionAgent(env)
 }
 
-func (dc *dataCollection) update() {
+func (dc *dataCollection) update(env cmds.Environment) {
 	var m runtime.MemStats
 	runtime.ReadMemStats(&m)
 
+	// get config info
+	var log = logging.Logger("analytics")
```
Comment (taiyangc) [inline analytics/analytics.go]: log should be at package level - see other loggers for example

**Pull request merged**